pkg/clients/network/network.go

```diff
@@ -132,8 +176,155 @@ func UpdateSubnetStatusFromAzure(v *v1alpha3.Subnet, az networkmgmt.Subnet) {
-	v.Status.Address = azure.ToString(az.IPAddress)
+	v.Status.AtProvider.State = azure.ToString(az.ProvisioningState)
+	v.Status.AtProvider.Etag = azure.ToString(az.Etag)
+	v.Status.AtProvider.ID = azure.ToString(az.ID)
+	v.Status.AtProvider.Address = azure.ToString(az.IPAddress)
+	v.Status.AtProvider.Version = string(az.PublicIPAddressVersion)
+	if az.IPConfiguration != nil {
+		v.Status.AtProvider.IPConfiguration = &v1alpha3.IPConfiguration{
+			PrivateIPAllocationMethod: string(az.IPConfiguration.PrivateIPAllocationMethod),
+			PrivateIPAddress:          az.IPConfiguration.PrivateIPAddress,
+			ProvisioningState:         azure.ToString(az.IPConfiguration.ProvisioningState),
+		}
+	}
+	if az.DNSSettings != nil {
+		v.Status.AtProvider.DNSSettings = &v1alpha3.PublicIPAddressDNSSettingsObservation{
+			DomainNameLabel: az.DNSSettings.DomainNameLabel,
+			FQDN:            az.DNSSettings.Fqdn,
+			ReverseFQDN:     az.DNSSettings.ReverseFqdn,
+		}
+	}
+}
+
+// LateInitializePublicIPAddress late-initilizes a PublicIPAddress resource
+func LateInitializePublicIPAddress(p *v1alpha3.PublicIPAddressProperties, in *networkmgmt.PublicIPAddress) {
+	p.PublicIPAddressDNSSettings = lateInitializeDNSSettings(p.PublicIPAddressDNSSettings, in.DNSSettings)
+	p.Tags = azure.LateInitializeStringMap(p.Tags, in.Tags)
+	if p.SKU == nil && in.Sku != nil {
+		p.SKU = &v1alpha3.SKU{
+			Name: string(in.Sku.Name),
+		}
+	}
+	if p.PublicIPPrefixID == nil && in.PublicIPPrefix != nil && in.PublicIPPrefix.ID != nil {
+		p.PublicIPPrefixID = in.PublicIPPrefix.ID
+	}
+	p.TCPIdleTimeoutInMinutes = azure.LateInitializeInt32PtrFromInt32Ptr(p.TCPIdleTimeoutInMinutes, in.IdleTimeoutInMinutes)
+	p.IPTags = lateInitializeIPTags(p.IPTags, in.IPTags)
+}
+
+func lateInitializeIPTags(t []v1alpha3.IPTag, from *[]networkmgmt.IPTag) []v1alpha3.IPTag {
+	if len(t) != 0 || from == nil || len(*from) == 0 {
+		return t
+	}
+	t = make([]v1alpha3.IPTag, len(*from))
+	for i, tag := range *from {
+		tag := tag
+		t[i] = v1alpha3.IPTag{
+			IPTagType: *tag.IPTagType,
+			Tag:       *tag.Tag,
+		}
+	}
+	return t
+}
+
+func lateInitializeDNSSettings(d *v1alpha3.PublicIPAddressDNSSettings, in *networkmgmt.PublicIPAddressDNSSettings) *v1alpha3.PublicIPAddressDNSSettings {
+	if in == nil {
+		return d
+	}
+	if d == nil {
+		d = &v1alpha3.PublicIPAddressDNSSettings{}
+	}
+	if d.DomainNameLabel == "" {
+		d.DomainNameLabel = azure.ToString(in.DomainNameLabel)
+	}
+	d.ReverseFQDN = azure.LateInitializeStringPtrFromPtr(d.ReverseFQDN, in.ReverseFqdn)
+	return d
+}
+
+// IsPublicIPAddressUpToDate is used to report whether given network.PublicIPAddress is in
+// sync with the PublicIPAddressProperties that the user desires.
+func IsPublicIPAddressUpToDate(p v1alpha3.PublicIPAddressProperties, in networkmgmt.PublicIPAddress) bool {
+	if !cmp.Equal(p.Tags, azure.ToStringMap(in.Tags), cmpopts.EquateEmpty()) {
+		return false
+	}
+
+	if !checkIPPrefixID(p.PublicIPPrefixID, in.PublicIPPrefix) {
```

Comment:
`check*` sounds too generic IMO. Do you think we can use something like `is*UpToDate` as a name or something that tells what the function actually checks?